packages/mdx/lib/plugin/recma-jsx-rewrite.js

```diff
@@ -363,18 +365,22 @@ export function recmaJsxRewrite(options) {
             }
 
             if (isNamedFunction(scope.node, '_createMdxContent')) {
-              for (const [
-                id,
-                componentName
-              ] of scope.idToInvalidComponentName) {
+              for (const id of [
+                ...scope.idToInvalidComponentName.keys()
+              ].sort()) {
                 // For JSX IDs that can’t be represented as JavaScript IDs (as in,
                 // those with dashes, such as `custom-element`), generate a
                 // separate variable that is a valid JS ID (such as `_component0`),
                 // and takes it from components:
                 // `const _component0 = _components['custom-element']`
                 declarations.push({
                   type: 'VariableDeclarator',
-                  id: {type: 'Identifier', name: componentName},
+                  id: {
+                    type: 'Identifier',
+                    name: /** @type {string} */ (
+                      scope.idToInvalidComponentName.get(id)
+                    )
+                  },
```

Comment:
It’s a bit sad this cast is needed, as we could use `.entries` above, though then that would make sorting a bit harder.

Reply:
Actually that’s not too bad. I decided to change it.